src/lib/components/ui/FilterBuilder.svelte

```diff
@@ -367,7 +484,30 @@
 							<label for="filter-value" class="block text-xs text-gray-600 dark:text-gray-400 mb-1"
 								>Value</label
 							>
-							{#if isSelectField}
+							{#if newFilterField === 'mapping_status'}
+								<!-- Special dropdown for mapping status -->
+								<select
+									id="filter-value"
+									bind:value={newFilterValue}
+									class="w-full rounded-md border border-gray-300 dark:border-gray-600 bg-white dark:bg-gray-700 text-gray-900 dark:text-white text-sm"
+								>
+									<option value="">Select status...</option>
+									<option value="planned">Planned</option>
+									<option value="implemented">Implemented</option>
+									<option value="verified">Verified</option>
```

Comment:
Same comment on shared type here, with that snippet of type code, this could be something like:
```ts
									{#each MAPPING_STATUS_OPTIONS as status}
										<option value={status.value}>{status.label}</option>
									{/each}
```